src/DotNetWorker.Core/Converters/DateTimeConverter.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Functions.Worker.Converters
+{
+    /// <summary>
+    /// Converter to bind DateTime/DateTime?/DateOnly/TimeOnly type parameters.
+    /// </summary>
+    internal sealed class DateTimeConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (
+                context.TargetType == typeof(DateTime) || context.TargetType == typeof(DateTime?)
```

Comment:
Should `DateTimeOffset` be handled as well?

Reply:
Good catch. Yes, we should support `DateTimeOffset` as well. I pushed a new iteration with that support.